---
title: "Dev Notes - Trials of Python"
date: 2019-02-24T15:58:26+08:00
lastmod: 2019-02-24T15:58:26+08:00
description: "Refactoring the website."
tags: ["News", "Programming", "JavaScript"]
categories: ["News"]
weight: 2
draft: true
toc:
  auto: false
math:
  enable: true
---

As part of the my "great refresh of 2021" project, I decided to update [joelvaneenwyk/docker-speedtest-analyser](https://github.com/joelvaneenwyk/docker-speedtest-analyser) which has required pretty minimal management over the years and has "just worked" for the most part. But alas, as my luck would have it, it finally stopped working which forced me to revisit the project. It has been 1.5+ years since I touched it, so it was time for a revamp. My goals:

1. Update to latest version of Alpine
2. S

### Python 3 and LIBC

```bash
    restart: unless-stopped
Speedtest root: '/usr/share/nginx/html'
--- Running speedtest using command line interface ---
Speedtest failed.
---------------------------------------
Traceback (most recent call last):
  File "/usr/share/nginx/html/scripts/runSpeedtest.py", line 38, in runSpeedtest
    test = Speedtest()
  File "/var/cache/nginx/.local/lib/python3.9/site-packages/speedtest.py", line 1095, in __init__
    self._opener = build_opener(source_address, timeout)
  File "/var/cache/nginx/.local/lib/python3.9/site-packages/speedtest.py", line 611, in build_opener
    opener.addheaders = [('User-agent', build_user_agent())]
  File "/var/cache/nginx/.local/lib/python3.9/site-packages/speedtest.py", line 680, in build_user_agent
    '(%s; U; %s; en-us)' % (platform.platform(),
  File "/usr/local/lib/python3.9/platform.py", line 1234, in platform
    libcname, libcversion = libc_ver()
  File "/usr/local/lib/python3.9/platform.py", line 196, in libc_ver
    with open(executable, 'rb') as f:
IsADirectoryError: [Errno 21] Is a directory: '/usr/share/nginx/html'
```
